web/dart.bzl

```diff
@@ -0,0 +1,101 @@
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Web Test rules for Dart."""
+
+load("@io_bazel_rules_dart//dart/build_rules:vm.bzl", "dart_vm_test")
+load("//web:web.bzl", "web_test_suite")
+
+
+def dart_web_test_suite(name,
+                        browsers,
+                        args=None,
+                        browser_overrides=None,
+                        config=None,
+                        dart_test_tags=("manual",),
+                        flaky=None,
+                        local=None,
+                        script_file=None,
+                        shard_count=None,
+                        size="large",
+                        tags=None,
+                        test_suite_tags=("manual",),
+                        timeout=None,
+                        visibility=None,
+                        web_test_data=None,
+                        **remaining_keyword_args):
+  """Defines a test_suite of web_test targets that wrap a dart_vm_test target.
+
+  Any attribute for dart_vm_test is supported by dart_web_test_suite. If the
+  argument isn't listed below, then the argument values have the same
+  exact meaning as in dart_vm_test.
+
+  Args:
+    name: The base name of the test.
+    browsers: A sequence of lavels specifying the browsers to use.
+    args: Args for web_test targets generated by this extension.
+    browser_overrides: Dictionary mapping from browsers to browser-specific
+      attributes, such as shard_count, flakiness, timeout, etc. For example:
+      {'chrome-linux': {'shard_count': 3, 'flaky': 1}
```

Comment:
Use example browser names from this project, e.g., '//browsers:chromium-native'.